Dockerfile

```diff
@@ -1,16 +1,59 @@
+# Support setting various labels on the final image
+ARG COMMIT=""
+ARG VERSION=""
+ARG BUILDNUM=""
+
 # Build Geth in a stock Go builder container
-FROM golang:1.16-alpine as builder
+FROM golang:1.17-alpine as builder
```

Comment:
We should modify `go.mod` first, and check it would not break any features.

Reply:
Dockerfile use alpine:latest image, lastest version is 3.16, but golang:1.16-alpine image use 3.15, two version not match, I will roll back this version

Reply:
Dockerfile rollback to alpine 3.15 base image, match golang 1.16 base image version

Reply:
Sorry to make a mistake, but Dmitry has already change the `go.mod` golang version to `1.17` in commit `d47998f3c799096dfccfde82b010249db3abcf7f`.
So the `golang:1.17-alpine` is reasonable.

Reply:
Currently, the only problem with golang version goes to another repo: `bas-genesis-config`, which is still using `go 1.16` in `go.mod`.
I'll open another PR for this go mod configuration.

Reply:
ok, I'll squash commits